src/test/ui/match/match-range-fail.stderr

```diff
@@ -19,6 +19,8 @@ LL |         10 ..= "what" => ()
 error[E0308]: mismatched types
   --> $DIR/match-range-fail.rs:17:9
    |
+LL |     match 5 {
+   |           - this match expression evaluates to `{integer}`
 LL |         'c' ..= 100 => { }
    |         ^^^^^^^^^^^ expected integer, found char
```

Comment:
This error (and https://github.com/rust-lang/rust/pull/57366/files#diff-eb11ffec25036e56175af396f96ddca0) should have a customized diagnostic that points at the range's components pointing out the difference.